### PR TITLE
[DOCFIX] Update guide on logging convention

### PR DESCRIPTION
**Pull request by apc999**
### What changes are proposed in this pull request?

Fix wrong documentation in logging convention

### Why are the changes needed?

Motivated by https://github.com/Alluxio/alluxio/pull/14278
Previous documentation on how to log is flawed

### Does this PR introduce any user facing changes?

N/A

### CODE REVIEW

**Comment by apc999** [comment]
@yuzhu can you help elaborate your comment?

File: docs/en/contributor/Code-Conventions.md [hunk 1]
```diff
@@ -268,15 +270,15 @@ trace may be found in debug level logs.
...
 
 ```java
-// Not recommended: this will print out the stack trace
+// Not recommended: wrong logging syntax 
+LOG.warn("Failed to do something: {}", e);
+// Not recommended: this will print out the stack trace, can be noisy
 LOG.warn("Failed to do something", e);
 // Not recommended: the exception class name is not included
-LOG.warn("Failed to do something", e.getMessage());
+LOG.warn("Failed to do something {}", e.getMessage());
```
[the remark this answers is not in the document]
Comment (yuzhu) [reply docs/en/contributor/Code-Conventions.md:281]: I think I meant to say that sometimes we print just the exception if it is INFO logging and we print the whole stacktrace when Debug logging is turned on.

**Review by yuzhu — APPROVED** [review]

File: docs/en/contributor/Code-Conventions.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (apc999) [reply docs/en/contributor/Code-Conventions.md:283]: done added some clarification

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**